features/proxyNew/state/proxyStateMachine.types.ts

```diff
@@ -0,0 +1,76 @@
+import { Observable } from 'rxjs'
+import { spawn } from 'xstate'
+
+import { TxHelpers } from '../../../components/AppContext'
+import { HasGasEstimation } from '../../../helpers/form'
+import { createProxyStateMachine } from './proxyStateMachine'
+
+export interface ProxyContext {
+  readonly txHelper: TxHelpers
+  readonly proxyAddress$: Observable<string | undefined>
+  readonly getGasEstimation$: (estimatedGasCost: number) => Observable<HasGasEstimation>
+  readonly safeConfirmations: number
+  gasData?: HasGasEstimation
+  txHash?: string
+  txError?: string
+  proxyConfirmations?: number
+  proxyAddress?: string
+}
+
+export type ProxyEvent =
+  | {
+      type: 'START'
+    }
+  | {
+      type: 'WAITING_FOR_APPROVAL'
+    }
+  | {
+      type: 'IN_PROGRESS'
+      proxyTxHash: string
+    }
+  | {
+      type: 'FAILURE'
+      txError?: string
+    }
+  | {
+      type: 'CONFIRMED'
+      proxyConfirmations?: number
+    }
+  | {
+      type: 'SUCCESS'
+      proxyAddress: string
+    }
+  | {
+      type: 'RETRY'
+    }
+  | {
+      type: 'GAS_COST_ESTIMATION'
+      gasData: HasGasEstimation
+    }
+
+export const PROXY_STAGES = [
+  'proxyIdle',
+  'proxyWaitingForApproval',
+  'proxyInProgress',
+  'proxyFailure',
+  'proxySuccess',
+] as const
+
+export type ProxyStages = typeof PROXY_STAGES[number]
```

Comment:
Could merge these two and remove unnecessary export.

```suggestion
export const PROXY_STAGES = [
  'proxyIdle',
  'proxyWaitingForApproval',
  'proxyInProgress',
  'proxyFailure',
  'proxySuccess',
] as const

type ProxyStateKeys = 'proxyIdle' |
  'proxyWaitingForApproval' |
  'proxyInProgress' |
  'proxyFailure' |
  'proxySuccess'
```

Also I'm not sure of the purpose of these types.  I can do this:

<img width="804" alt="image" src="https://user-images.githubusercontent.com/3783198/181041938-a14297af-7879-462b-8d6b-3e2b4f8c85dd.png">

and it compiles fine.  What is this type for?

Reply:
you can, it is an issue with types in the library I think. 
types are useful when you are checking the state of the machine:
```
state.matches('proxyIdle'):
```

FYI: unnecessary export removed.